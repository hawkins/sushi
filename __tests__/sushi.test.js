const { Roll } = require("../src/sushi");

describe("Sushi Roll", () => {
  it("can be made from ingredients", () => {
    expect(() => {
      new Roll([
        "white rice",
        "seaweed wrap",
        "fried shrimp",
        "avocado",
        "cream cheese",
        "shredded carrot",
        "green bean",
        "spicy mayo",
        "sweet chili"
      ]);
    }).not.toThrowError();
  });

  it("can't be made from random junk'", () => {
    expect(() => {
      new Roll([
        "tonka truck",
        "lego brick",
        "your neighbor's garage door opener",
        "frozen water",
        "caffeine pills",
        "bike tire"
      ]);
    }).toThrowErrorMatchingSnapshot();
  });

  it("can be described as a map of ingredients to their presence", () => {
    expect(
      new Roll([
        "white rice",
        "seaweed wrap",
        "fried shrimp",
        "avocado",
        "cream cheese",
        "shredded carrot",
        "green bean",
        "spicy mayo",
        "sweet chili"
      ]).getDescriptor()
    ).toMatchSnapshot();
  });
});
