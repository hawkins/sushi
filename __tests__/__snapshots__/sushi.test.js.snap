// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sushi Roll can be described as a map of ingredients to their presence 1`] = `
Object {
  "apple": 0,
  "avocado": 1,
  "brown rice": 0,
  "crab": 0,
  "cream cheese": 1,
  "fried shrimp": 1,
  "green bean": 1,
  "raw salmon": 0,
  "seaweed wrap": 1,
  "shredded carrot": 1,
  "spicy mayo": 1,
  "sweet chili": 1,
  "sweet soy": 0,
  "tofu wrap": 0,
  "white rice": 1,
}
`;

exports[`Sushi Roll can't be made from random junk' 1`] = `"Sushi Roll ingredients must be a subset of ALL_INGREDIENTS"`;
